### PR TITLE
elasticsearch_visitor: handle malformed queries

### PR DESCRIPTION
**Pull request by chris-asl**
* Add support for malformed queries and also, queries with malformed
  part (along with their recognized parse tree).

Signed-off-by: Chris Aslanoglou <chris.aslanoglou@gmail.com>

### CODE REVIEW

File: inspire_query_parser/visitors/elastic_search_visitor.py [hunk 1]
```diff
@@ -96,6 +98,33 @@ def visit_value_query(self, node):
...
+            }
+
+        else:
+            return {
+                "bool": {
+                    "must": [node.left.accept(self)],
+                    "should": [node.right.accept(self)]
+                }
+            }
+
```
Comment (ammirate) [inline inspire_query_parser/visitors/elastic_search_visitor.py]: what about...?
```python
query = {
    "bool": {
        "must": [
            node.left.accept(self),
        ],
    }
}

# handling malformed query in according to parser config
if DEFAULT_ES_OPERATOR_FOR_MALFORMED_QUERIES == ES_MUST_QUERY:
    query['bool']['must'].append(node.right.accept(self))
else:
   query['bool']['should'] = [node.right.accept(self)]
```

File: tests/test_elastic_search_visitor.py [hunk 2]
```diff
@@ -23,25 +23,28 @@
...
-from inspire_query_parser.utils.format_parse_tree import emit_tree_format
 from inspire_query_parser.visitors.elastic_search_visitor import \
     ElasticSearchVisitor
 from inspire_query_parser.visitors.restructuring_visitor import \
     RestructuringVisitor
 
 
 def _parse_query(query_str):
-    print("Parsing: " + query_str)
+    # print("Parsing: " + query_str)
```
Comment (ammirate) [inline tests/test_elastic_search_visitor.py]: leftover?

File: tests/test_elastic_search_visitor.py [hunk 3]
```diff
@@ -23,25 +23,28 @@
...
     stateful_parser = StatefulParser()
     restructuring_visitor = RestructuringVisitor()
     elastic_search_visitor = ElasticSearchVisitor()
     _, parse_tree = stateful_parser.parse(query_str, parser.Query)
     parse_tree = parse_tree.accept(restructuring_visitor)
-    print("RST: \n" + emit_tree_format(parse_tree))
+    # print("RST: \n" + emit_tree_format(parse_tree))
```
Comment (ammirate) [inline tests/test_elastic_search_visitor.py]: also here? :smiley:

File: inspire_query_parser/visitors/elastic_search_visitor.py (hunk 1 above)
Comment (chris-asl) [reply inspire_query_parser/visitors/elastic_search_visitor.py]: I prefer this one, will amend!

**Comment by chris-asl** [comment]
Regarding setting the config variable, is in `config.py`, [here](https://github.com/inspirehep/inspire-query-parser/pull/23/files#diff-1e5e3f0f9cc804a33297890bf000941dR241).

**Pull request merged**